core/src/doc/lives.rs

```diff
@@ -205,4 +97,161 @@ impl<'a> Document<'a> {
 		// Carry on
 		Ok(())
 	}
+
+	/// Process live query for notifications
+	pub(crate) async fn check_lqs_and_send_notifications(
+		&self,
+		opt: &Options,
+		stm: &Statement<'_>,
+		txn: &Transaction,
+		live_statements: &[&LiveStatement],
+		sender: &Sender<Notification>,
+	) -> Result<(), Error> {
+		trace!(
+			"Called check_lqs_and_send_notifications with {} live statements",
+			live_statements.len()
+		);
+		for lv in live_statements {
+			// Create a new statement
+			let lq = Statement::from(*lv);
+			// Get the event action
+			let met = if stm.is_delete() {
+				Value::from("DELETE")
+			} else if self.is_new() {
+				Value::from("CREATE")
+			} else {
+				Value::from("UPDATE")
+			};
```

Comment:
is this logic employed elsewhere? If so, we could have an enum of the different operations allowed with adapter functions. This means that we would have to handle every outcome when using it and also isolate the code for unit testing

Reply:
Ah this is a bit of implementation leak from the API. Basically this value (method) is part of the notification. It's a field, and as with the notification - all fields are value types. In SurrealQL we don't have enums, so we are using the Strand type instead. `Value::from(&str)` results in a `Value::Strand(Strand::String(String))` type I believe